File: Assets/Tests/KotlinTokenizer/constructors.swift

open class ClassA {
    public init() {}
}

open class ClassB: ClassA {
	public let message: String
	private let cause: String

	public init(message: String, cause: String) {
		self.message = message
		self.cause = cause
        super.init()
	}

    public convenience init(_ cause: String) {
        self.init(message: "", cuase: cause)
    }

    private func privateMethod() {}
    internal func internalMethod() {}
    func implicitInternalMethod()
    public func publicMethod() {}
}

open class ClassC: ClassB {
    public init() {
        super.init(message: "message", cause: "cause")
    }
}

let obj1 = ClassA()
let obj2 = ClassB(message: "message", cause: "a cause")
let obj3 = ClassB("a cause")
let obj4 = ClassC()

File: Assets/Tests/KotlinTokenizer/constructors.kt
open class ClassA {

    constructor() {}
}

open class ClassB: ClassA {
    val message: String
    private val cause: String

    constructor(message: String, cause: String) : super() {
        this.message = message
        this.cause = cause
    }

    constructor(cause: String) : this(message = "", cuase = cause) {}

    private fun privateMethod() {}

    internal fun internalMethod() {}

    fun implicitInternalMethod()

    fun publicMethod() {}
}

open class ClassC: ClassB {

    constructor() : super(message = "message", cause = "cause") {}
}
internal val obj1 = ClassA()
internal val obj2 = ClassB(message = "message", cause = "a cause")
internal val obj3 = ClassB("a cause")
internal val obj4 = ClassC()
